projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -18,3 +20,212 @@ function jetpack_blogging_prompts_add_meta_data( $keys ) {
 	$keys[] = '_jetpack_blogging_prompt_key';
 	return $keys;
 }
+
+add_filter( 'rest_api_allowed_public_metadata', 'jetpack_blogging_prompts_add_meta_data' );
+
+/**
+ * Sets up a new post as an answer to a blogging prompt.
+ *
+ * Called on `wp_insert_post` hook.
+ *
+ * @param int $post_id ID of post being inserted.
+ * @return void
+ */
+function jetpack_setup_blogging_prompt_response( $post_id ) {
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
We are generating the link to be clicked from a WP.com notification, so it's possible the nonce would change before the user clicks the link. I'll add a comment.

Not ideal, as we are "setting a field," in the form of a tag, post_meta with the prompt id, and a prefilled pull quote block, though no data is saved until the user publishes. Please let me know if you have other thoughts or a better solution!

Reply:
As long as it's well-explained in the comment, I think that's a good start. :+1:

Do I have this right? There's a link in the notification, and clicking the link winds up calling this function. You can't put a nonce in the link, because it might be a while in between when the link is generated and when it's clicked. The "attack" that a nonce would prevent would be some third party tricking the blog admin's browser into following the link, maybe repeatedly to flood the site with these posts. Ideally the comment should address why that potential attack isn't something to actually worry about.